### PR TITLE
Implements rspec configuration test

### PR DESCRIPTION
**Pull request by ChrisBr**

### CODE REVIEW

File: src/api/spec/controllers/webui/configuration_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+RSpec.describe Webui::ConfigurationController do
+  let(:confirmed_user) { create(:confirmed_user) }
+  let(:admin_user) { create(:admin_user) }
+
+  describe 'GET #index' do
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/configuration_controller_spec.rb]: I don't like this. We should test that this controller has `before_filter :require_admin` and that Webui::WebuiController#require_admin works

File: src/api/spec/models/configuration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    expect(Configuration.count).to eq(0)
+    Configuration.title
+    expect(Configuration.count).to eq(1)
+  end
+
+  it 'uses the first Configuration if there is one' do
+    # The first configuration is created by db/seeds.rb
+    expect(Configuration.count).to eq(1)
+    Configuration.title
+    expect(Configuration.count).to eq(1)
```
Comment (hennevogel) [inline src/api/spec/models/configuration_spec.rb]: What does the count verify?

File: src/api/spec/controllers/webui/configuration_controller_spec.rb (hunk 1 above)
Comment (ChrisBr) [inline src/api/spec/controllers/webui/configuration_controller_spec.rb]: Yep, will do that with a shoulda matcher and test the before filter in the webui controller test!

File: src/api/spec/models/configuration_spec.rb (hunk 2 above)
Comment (ChrisBr) [inline src/api/spec/models/configuration_spec.rb]: That we still have only one Configuration after calling the method_missing method

**Comment by mdeniz** [comment]
LGTM :+1:

File: src/api/spec/models/configuration_spec.rb (hunk 2 above)
Comment (hennevogel) [inline src/api/spec/models/configuration_spec.rb]: Okay, pretty misleading description then.

**Comment by ChrisBr** [comment]
@hennevogel @mdeniz updated!

**Comment by ChrisBr** [comment]
Okay, adapted the PR a little bit:
- reworded the model tests
- added a separate test for require_admin
- test that the controller uses the require_admin before filter

@hennevogel @mdeniz please review again!

File: src/api/spec/controllers/webui/configuration_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    it { should use_before_action(:require_admin) }
+
+    it 'is allowed as admin User' do
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/configuration_controller_spec.rb]: same here, why do you test this? Don't you trust your own `:require_admin` test? ;)

Comment (ChrisBr) [inline src/api/spec/controllers/webui/configuration_controller_spec.rb]: I do trust it but I also want to test the functionality of the actions (and this is only possible as admin). We can discuss about renaming the test, so the name isn't focused on the admin privileges.

Comment (hennevogel) [inline src/api/spec/controllers/webui/configuration_controller_spec.rb]: And the functionality of the index method of the configuration controller is what?

**Comment by hennevogel** [comment]
Awesome, way better!

**Pull request merged**